cl/disclosures/tasks.py

```diff
@@ -362,60 +314,53 @@ def save_disclosure(extracted_data: dict, disclosure) -> None:
 
-    :param data: File data
-    :return: URL or first URL on AWS
-    """
-    if data["disclosure_type"] in ["jw", "single", "jef", "pdf"]:
-        url = data["url"]
-    else:
-        url = data["urls"][0]
-    return url
-
-
-def get_disclosure_from_pdf_path(disclosure_url: str):
-    """Convenience method to get disclosure from download filepath
-
-    :param disclosure_url: The URL of the first link (if there are more than
-    one) of the source FD tiff(s)/PDF
-    :return: Financial Disclosure object
-    """
-    return FinancialDisclosure.objects.get(download_filepath=disclosure_url)
+    sha1_hash = sha1(response.content)
+    if FinancialDisclosure.objects.filter(sha1=sha1_hash).exists():
+        logger.error(
+            "PDF already in system.",
+            extra={"disclosure_id": disclosure_key},
+        )
+        interface.delete(disclosure_key)
+        return
 
+    page_count = microservice(
+        service="page-count",
+        filename=data["url"].split("/")[-1],
+        file=response.content,
+    ).content
+    if not page_count:
+        logger.error(
+            msg=f"Page count failed",
+            extra={"disclosure_id": disclosure_key, "url": data["url"]},
+        )
+        interface.delete(disclosure_key)
+        return
 
-def generate_or_download_disclosure_as_pdf(
-    data: Dict[str, Union[str, int, List[str]]],
-    pdf_url: Optional[str],
-) -> requests.Response:
-    """Generate or download PDF content from images or urls.
+    # Make disclosure
+    disclosure = FinancialDisclosure(
+        year=int(data["year"]),
+        page_count=page_count,
+        person=Person.objects.get(id=data["person_id"]),
+        sha1=sha1_hash,
+        has_been_extracted=False,
+        report_type=data.get("report_type", -1),
+        download_filepath=data.get("url"),
+    )
 
-    :param data: Data to process.
-    :param pdf_url: The URL of PDF in S3
-    :return: Response containing PDF
-    """
-    if pdf_url:
-        logger.info(f"Downloading PDF: {pdf_url}")
-        return requests.get(pdf_url, timeout=60 * 20)
-    elif data["disclosure_type"] == "jw":
-        logger.info(f"Downloading JW PDF: {quote(data['url'], safe=':/')}")
-        return requests.get(data["url"], timeout=60 * 20)
-    elif data["disclosure_type"] == "jef":
-        logger.info(f"Downloading JEF PDF: {quote(data['url'], safe=':/')}")
-        return requests.get(data["url"], timeout=60 * 20)
-    elif data["disclosure_type"] == "single":
-        urls = [data["url"]]
-    elif data["disclosure_type"] == "pdf":
-        return requests.get(data["url"], timeout=60 * 20)
-    else:
-        urls = data["urls"]
-    logger.info(f"Processing url:{quote(urls[0], safe=':/')}")
-    return requests.post(
-        settings.BTE_URLS["images-to-pdf"]["url"],
-        json=json.dumps({"urls": urls}),
-        timeout=settings.BTE_URLS["images-to-pdf"]["timeout"],
+    # Save and upload & generate thumbnail
```

Comment:
Did I miss where the thumbnail is getting generated or does this comment need to be updated?